Modules/Sources/Yosemite/Tools/POS/POSCatalogPersistenceService.swift

```diff
@@ -154,6 +156,22 @@ final class POSCatalogPersistenceService: POSCatalogPersistenceServiceProtocol {
     }
 }
 
+private extension POSCatalogPersistenceService {
+    /// Filters out variations whose parent products are not in the catalog.
+    /// This can happen when the API returns public variations but their parent products are not public.
+    func filterOrphanedVariations(from catalog: POSCatalog) -> POSCatalog {
+        let productIDs = catalog.products.map { $0.productID }
+        let variations = catalog.variations.filter { variation in
+            let parentExists = productIDs.contains { $0 == variation.productID }
+            if !parentExists {
+                DDLogWarn("Variation \(variation.productVariationID) references missing product \(variation.productID)")
```

Comment:
Good thinking logging this.
```suggestion
                DDLogWarn("Variation \(variation.productVariationID) references missing product \(variation.productID) - it will not be available in POS.")
```

Reply:
Updated in https://github.com/woocommerce/woocommerce-ios/pull/16301/commits/786a89a006186a51601fdb58c57f96f8b0711d96.